### PR TITLE
Fix Tensor Device Inconsistencies in PGD

### LINKED ISSUE #2123: Tensor Device Inconsistencies in Projected Gradient Descent Algorithm.
**Describe the bug**
Then tensors are not on the same device.

**To Reproduce**
1. Go to 'art/evasion/projected_gradient_descent/'
2. Click on 'projected_gradient_descent_pytorch.py'
3. Scroll down to 'line 272'

**Expected behavior**
Expected all tensors to be on the same device.

**Screenshots**
Change
image
To
image

**System information (please complete the following information):**
 - OS Linux
 - Python version 3.7.9
 - ART version or commit number 1.14.0
 - Pytorch 1.11.0+cu113

### PR DESCRIPTION
**Pull request by SignedQiu**
# Description

Please include a summary of the change, motivation and which issue is fixed. Any dependencies changes should also be included.

Fixes #2123

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [ ] Test A
- [ ] Test B

**Test Configuration**:
- OS
- Python version
- ART version or commit number
- TensorFlow / Keras / PyTorch / MXNet version

# Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by beat-buesser** [comment]
Hi @SignedQiu Thank you very much for you pull request! Could you please sign off your commit to pass the DCO check above?

To add your Signed-off-by line to every commit in this branch:

- Ensure you have a local copy of your branch by [checking out the pull request locally via command line](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/checking-out-pull-requests-locally).
- In your local branch, run: `git rebase HEAD~1--signoff`
- Force push your changes to overwrite the branch: `git push --force-with-lease origin signedqiu`

**Review by beat-buesser — APPROVED** [review]
Hi @SignedQiu Thank you very much for fixing this issue and congratulations to your first PR accepted to ART!
